pkg/platform/model/buildplanner/buildscript.go

```diff
@@ -2,18 +2,41 @@ package buildplanner
 
 import (
 	"encoding/json"
+	"fmt"
+	"os"
 	"strings"
 	"time"
 
+	"github.com/ActiveState/cli/internal/constants"
 	"github.com/ActiveState/cli/internal/errs"
 	"github.com/ActiveState/cli/internal/logging"
-	"github.com/ActiveState/cli/internal/rtutils/ptr"
+	"github.com/ActiveState/cli/internal/multilog"
 	"github.com/ActiveState/cli/pkg/buildscript"
 	"github.com/ActiveState/cli/pkg/platform/api/buildplanner/request"
 	bpResp "github.com/ActiveState/cli/pkg/platform/api/buildplanner/response"
+	"github.com/ActiveState/cli/pkg/projectfile"
 )
 
-func (b *BuildPlanner) GetBuildScript(commitID string) (*buildscript.BuildScript, error) {
+func buildScriptCheckoutInfo(owner, project, branch, commitID string, atTime time.Time) *buildscript.CheckoutInfo {
+	// Note: cannot use api.GetPlatformURL() due to import cycle.
+	host := constants.DefaultAPIHost
+	if hostOverride := os.Getenv(constants.APIHostEnvVarName); hostOverride != "" {
+		host = hostOverride
+	}
+	pjf := projectfile.NewProjectField()
+	err := pjf.LoadProject(fmt.Sprintf("https://%s/%s/%s", host, owner, project))
```

Comment:
Nit: might be nice to add the host stuff directly to the `NewProjectField()` constructor. Then just use the setters, and you don't need `LoadProject` here at all.